experiment/autobumper/bumper/bumper.go

```diff
@@ -518,14 +518,59 @@ func GitCommitAndPush(remote, remoteBranch, name, email, message string, stdout,
 }
 
 // GitPush push the changes to the given remote and branch.
-func GitPush(remote, remoteBranch string, stdout, stderr io.Writer) error {
+func GitPush(remote, remoteBranch string, stdout, stderr io.Writer, workingDir string) error {
 	logrus.Info("Pushing to remote...")
-	if err := Call(stdout, stderr, "git", "push", "-f", remote, fmt.Sprintf("HEAD:%s", remoteBranch)); err != nil {
+	gitArgs := []string{}
+	if workingDir != "" {
+		gitArgs = append(gitArgs, fmt.Sprintf("--git-dir=%s/.git", workingDir), fmt.Sprintf("--work-tree=%s", workingDir))
+	}
+	gitPushArgs := append(gitArgs, "push", "-f", remote, fmt.Sprintf("HEAD:%s", remoteBranch))
+	if err := Call(stdout, stderr, "git", gitPushArgs...); err != nil {
 		return fmt.Errorf("failed to git push: %w", err)
 	}
 	return nil
 }
 
+// RunAndCommitIfNeeded makes a commit in the workingDir if there are
+// any changes resulting from the command execution. Returns true if a commit is made
+func RunAndCommitIfNeeded(stdout, stderr io.Writer, author, cmd string, args []string, workingDir string) (bool, error) {
+	fullCommand := fmt.Sprintf("%s %s", filepath.Base(cmd), strings.Join(args, " "))
```

Comment:
Same comment like I had in the ci-tools PR - `fullCommand` is used in logging. While reading logs, I want to see what the tool really does (calls `cmd`) without prettyfing it with `filepath.Base`.